apps/web2/src/pages/PlanAnEventPage/PlanAnEventPage.tsx

```diff
@@ -0,0 +1,353 @@
+  const { user } = useUserContext();
+  const { userRecord } = useUserRecordContext();
+  const history = useHistory();
+  const [invitees, updateInvitees] = React.useState<Email[]>([]);
+  const recaptchaRef = React.useRef<ReCAPTCHA>(null);
+  const inputRef = React.useRef<HTMLInputElement>();
+
+  const currentDate = moment().format('YYYY-MM-DD');
+  const [startTimeValue, setStartTimeValue] = React.useState<moment.Moment>(
+    moment().startOf('hour')
+  );
+  const [endTimeValue, setEndTimeValue] = React.useState<moment.Moment>(
+    moment().startOf('hour').add(15, 'minutes')
+  );
+
+  const onSubmitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    // Fail immediately if no token
+    assert(recaptchaRef.current, 'ReCAPTCHA has not loaded');
+    const token = await recaptchaRef.current.executeAsync();
+    assert(token, 'Missing ReCAPTCHA token');
+
+    const formData = new FormData(e.target as HTMLFormElement);
+    const formValues = Object.fromEntries(
+      formData.entries()
+    ) as unknown as EventPlanInfo;
+
+    // We always convert to 24 hour time when storing these time
+    // intervals in firestore
+    formValues.dailyStartTime = moment(
+      formValues.dailyStartTime,
+      SUPPORTED_TIME_FORMATS
+    ).format(TIME_FORMAT.TWENTY_FOUR_HOURS);
+    formValues.dailyEndTime = moment(
+      formValues.dailyEndTime,
+      SUPPORTED_TIME_FORMATS
+    ).format(TIME_FORMAT.TWENTY_FOUR_HOURS);
+
+    assert(
+      moment(formValues.dailyStartTime, SUPPORTED_TIME_FORMATS, true).isValid()
+    );
+    assert(
+      moment(formValues.dailyEndTime, SUPPORTED_TIME_FORMATS, true).isValid()
+    );
+
+    const eventPlanData: EventPlanInfo & {
+      invitees: Email[];
+      'g-recaptcha-response': string;
+    } = {
+      ...(formValues as EventPlanInfo),
+      invitees,
+      'g-recaptcha-response': token,
+    };
+
+    console.log(eventPlanData);
+
+    const {
+      data: {
+        data: [eventPlanId],
+      },
+    } = await api.post('/event-plans/create', JSON.stringify(eventPlanData));
+
+    console.log('Event plan created: ', eventPlanId);
+
+    history.push(`/event-plans/${eventPlanId}`);
+  };
+
+  const AddGuestHandler = () => {
+    if (inputRef.current) {
+      const newGuestUID = inputRef.current.value;
+      updateInvitees([...invitees, newGuestUID]);
+      inputRef.current.value = '';
+    }
+  };
+
+  const onRemoveHandler = (index: number) => {
+    // removes item at index and modifies guests in place
+    invitees.splice(index, 1);
+    updateInvitees([...invitees]);
+  };
+
+  return (
+    <Sidebar>
+      <main className="max-w-lg mx-auto pt-20 pb-12 px-4 lg:pb-16">
+        <form onSubmit={onSubmitHandler}>
+          <input type="hidden" name="hostId" value={user?.uid} />
+          <div className="space-y-6">
+            <div>
+              <h1 className="text-lg leading-6 font-medium text-gray-900">
+                Event Settings
```

Comment:
Not sure if you meant to change this, but it seems like it should be something else.